src/main/java/com/softserve/messaging/service/MailServiceImpl.java

```diff
@@ -12,22 +12,23 @@
 
 @Service
 @RequiredArgsConstructor
-@Log4j2
+@Slf4j
 public class MailServiceImpl implements MailService {
 
     private final JavaMailSender javaMailSender;
 
-    public void sendReceiptMessage(UserReceipt userReceipt) {
+    public void sendMessage(MailTemplate mailTemplate) {
         MimeMessage message = javaMailSender.createMimeMessage();
 
         try {
             MimeMessageHelper helper = new MimeMessageHelper(message, true);
 
-            helper.setTo(userReceipt.getMail());
-            helper.setSubject(UserReceipt.TITLE);
-            helper.setText(userReceipt.toString());
+            helper.setTo(mailTemplate.getMail());
+            helper.setSubject(mailTemplate.getTopic());
+            helper.setText(mailTemplate.getBody());
 
             javaMailSender.send(message);
+
             log.info("Mail was successfully sent");
         } catch (MessagingException e) {
             log.error("Mail wasn't sent, exception: {}", e.getMessage());
```

Comment:
it might not contain the needed message the cause of`MessagingException` is a nested exception, `toString` will fix that

Reply:
for the future: this message shouldn't be lost. In case of exception, it should be saved to DLQ topic and replayed later.